turbo/jsonrpc/tracing.go

```diff
@@ -224,7 +224,12 @@ func (api *PrivateDebugAPIImpl) traceBlock(ctx context.Context, blockNrOrHash rp
 	}
 
 	if dbg.AssertEnabled {
-		if block.GasUsed() != usedGas {
+		var refunds = true
+		if config.NoRefunds != nil && *config.NoRefunds {
```

Comment:
why this flag only affects behavior inside `Assert` section? (which disabled by default)

Reply:
In the sw the refunds flag is managed there is this extra check that fail the RPC-API request in case of refunds = false
I.e in debug_traceBlockByNumber() the call stack is
- TraceBlockByNumber(config)
- traceBlock(... config, ..)
- transactions.TraceTx(.., config, ..)
- ExecuteTraceTx(config,..) -> according config assign refunds variable and calls ApplyMessage
- ApplyMessage(evm *vm.EVM, msg Message, gp *GasPool, refunds bool, gasBailout bool) 
- TransitionDb(...refunds,..)
Where in case of refunds make or not refunds to the caller
 if refunds && !gasBailout {
                if rules.IsLondon {
                        // After EIP-3529: refunds are capped to gasUsed / 5
                        st.refundGas(params.RefundQuotientEIP3529)
                } else {
                        // Before EIP-3529: refunds were capped to gasUsed / 2
                        st.refundGas(params.RefundQuotient)
                }
        }

Reply:
seems we need this: https://github.com/erigontech/erigon/pull/13237
and your fix also

Reply:
please add `ERIGON_ASSERT=true` to some your RPC server.